components/templates/src/global_fns.rs

```diff
@@ -240,12 +247,149 @@ pub fn make_resize_image(imageproc: Arc<Mutex<imageproc::Processor>>) -> GlobalF
     })
 }
 
+/// A global function to load data from a data file.
+/// Currently the supported formats are json, toml and csv
+pub fn make_load_data(content_path: PathBuf) -> GlobalFn {
+    Box::new(move |args| -> Result<Value> {
+        let path_arg: String = required_arg!(
+            String, 
+            args.get("path"), 
+            "`load_data`: requires a `path` argument with a string value, being a path to a file"
+        );
+        let kind_arg = optional_arg!(
+            String,
+            args.get("kind"),
+            "`load_data`: `kind` needs to be an argument with a string value, being one of the supported `load_data` file types (csv, json, toml)"
+        );
+
+        let full_path = content_path.join(&path_arg);
+
+        let extension = match full_path.extension() {
+            Some(value) => value.to_str().unwrap().to_lowercase(),
+            None => return Err(
+                Error::from_kind(
+                    ErrorKind::Msg(
+                        format!("`load_data`: Cannot parse file extension of specified file: {}", path_arg).into()
+                        )
+                    )
+                )
+        };
+
+        let file_kind = kind_arg.unwrap_or(extension);
+
+        let result_value: Result<Value> = match file_kind.as_str() {
+            "toml" => load_toml(&full_path),
+            "csv" => load_csv(&full_path),
+            "json" => load_json(&full_path),
+            _ => Err(
+                Error::from_kind(
+                    ErrorKind::Msg(
+                        String::from(
+                            format!("'load_data': {} - is an unsupported file kind", file_kind)
+                            )
+                        )
+                    )
+                )
+        };
+
+        result_value
+    })
+}
+
+/// load/parse a json file from the given path and place it into a 
+/// tera value
+fn load_json(json_path: &PathBuf) -> Result<Value> {
+    let content_string: String = read_file(json_path)
+        .map_err(|err: errors::Error| {
```

Comment:
that can be changed to 
```
.chain_err(|| format!("`load_data`: error loading json file"."))
```

I think
Right now it seems to not display the original error